node/forest_libp2p/src/behaviour.rs

```diff
@@ -56,7 +56,7 @@ use tiny_cid::Cid as Cid2;
 
 lazy_static! {
     static ref VERSION: &'static str = env!("CARGO_PKG_VERSION");
-    static ref CURRENT_COMMIT: &'static str = git_version!();
+    static ref CURRENT_COMMIT: &'static str = git_version!(fallback = "unknown");
```

Comment:
I'm not sure we want to lose that information. What if we would simply copy `.git/` into the build phase of the image?

Reply:
The Docker build should definitely retain the `.git` information. There is currently no plan to distribute this code without git references.

Reply:
After build the source directory seems to be removed:
```
$ docker exec -it 280c531fdb26 bash
root@280c531fdb26:/# cd /usr/src/forest/
bash: cd: /usr/src/forest/: No such file or directory
root@280c531fdb26:/# cd /usr/src/       
root@280c531fdb26:/usr/src# ls -lah
total 12K
drwxr-xr-x 2 root root 4.0K Oct  3 09:00 .
drwxr-xr-x 1 root root 4.0K Feb 19 11:30 ..
root@280c531fdb26:/usr/src# cd ..
root@280c531fdb26:/usr# cd ..
root@280c531fdb26:/# find ./ -name "rpc-client" | wc -l
0
```
but I will study the build process what happens with the git repository.

Often it is exclude from the build process because it can become extensive.

For the case the commit information is strictly necessary I would suggest to export it into a kind of persistent _release version file_ and then read it from there.

Reply:
As commented before the "`.git/" directory is excluded from the image build:
[excluded files from image build](https://github.com/ChainSafe/forest/blob/e66d0f9e2f810752bd3330d9ae2370f7197b234b/.dockerignore#L2)
like commented before the _Git_ version information can grow huge on old projects and will slow down the build process very much.

Reply:
I believe it's completely fine to include `.git/` given we have a multi-stage build. It shouldn't add any more overhead than the initial clone and the release image won't have it anyway.

Reply:
The `git_version!()` macro always panicks on any error. 
As documented at
[Replace `git-version` by `forest_git_utils`](https://github.com/ChainSafe/forest/issues/1384#issuecomment-1107849388)
a serialized file can provide the missing _Git_ information in `config_forest_commit.toml` in case the _Git_ Command fails.

Reply:
The test documents that the _Git_ Reference is correctly included into the _Forest_ Service Version:
https://github.com/bodo-hugo-barwich/forest/runs/6147904955?check_suite_focus=true#step:7:11
```
 Docker: Service 'forest' Version:
forest v0.2.2/unstable/mainnet/d24c6014
Git: 'CURRENT_COMMIT' serialized:
d24c6014
```